yawl/lib/yawl/widget/ToggleSwitch.lua

```diff
@@ -0,0 +1,70 @@
+local gpu = require("component").gpu
+local class = require("libClass2")
+local Frame = require("yawl.widget.Frame")
+local Text = require("yawl.widget.Text")
+local Rectangle = require("yawl.widget.Rectangle")
+
+---@class ToggleSwitch:Widget
+---@field private _size Size
+---@operator call:ToggleSwitch
+---@overload fun(parent:Frame,x:number,y:number,width:number,height:number,backgroundColor:number)
+local ToggleSwitch = class(Frame)
+
+---Create a new ToggleSwitch
+---@param parent Frame
+---@param x number
+---@param y number
+---@param width number
+---@param height number
+---@param backgroundColor number
+---@return ToggleSwitch
+function ToggleSwitch:new(parent, x, y, width, height, backgroundColor, foregroundColor)
+    checkArg(1, parent, 'table')
+    checkArg(2, x, 'number')
+    checkArg(3, y, 'number')
+    checkArg(4, width, 'number')
+    checkArg(5, height, 'number')
+    checkArg(6, backgroundColor, 'number','nil')
+    checkArg(6, foregroundColor, 'number','nil')
+    local o = self.parent(parent, x, y)
+    setmetatable(o, {__index = self})
+    o._size = {width = 1, height = 1}
+    o._value = false
+    o._visual = Rectangle(o, 1, 1, math.min(width, height), height, foregroundColor or 0)
```

Comment:
Since the widget is a Frame, the rectangle is already accessible in `self._childs[1]`

Reply:
the rectangle in here should be its only child and its better to store it as _visual (might change to "_slider" than assuming its the first index in its child table